### PR TITLE
Add dfg as a pre-commit hook

### PR DESCRIPTION
**Pull request by vyasr**
## Description

This change allows local and remote runs to handle calls to dfg identically, and removes the need for a separate CI check.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vyasr** [comment]
Converting to draft pending some discussions on https://github.com/rapidsai/shared-action-workflows/issues/46.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -147,6 +147,10 @@ repos:
                     ^CHANGELOG.md$|
                     ^.*versioneer.py$
                   )
+      - repo: https://github.com/rapidsai/dependency-file-generator
+        rev: v1.2.0
```
Comment (vyasr) [inline .pre-commit-config.yaml:151]:
```suggestion
        rev: v1.4.0
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -147,6 +147,10 @@ repos:
...
+        hooks:
+            - id: rapids-dependency-file-generator
```
Comment (vyasr) [inline .pre-commit-config.yaml:153]:
```suggestion
            - id: rapids-dependency-file-generator
              args: ["--clean"]
```

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**